### DISCUSSION TITLE
Exclude rpyc files from the workspace folder list.

### DISCUSSION THREAD

**Discussion by Savage1969** [renpy/vscode-language-renpy#147] [Q&A] [answered]
VSC: v1.68.1
RenPy Extension: v2.0.10

Verified the default "Exclude Rpyc" is checked in both User & Workspace but yet those files are still in the list in every project I open. What's the trick I'm missing as I'm fairly certain it's not a product issue but something very simple I'm missing. Thank you and regards.

**Comment by duckdoom4** [comment, marked as answer]
Let me have a quick look. Maybe it is an issue in the code

**Reply by duckdoom4** [reply]
Alright, it works fine on my end.

Can you verify that you have the file `.vscode/settings.json` in the root of your workspace.
And that the file contains:

```
{
  "files.exclude": {
    "**/*.rpyc": true
  }
}
```

![image](https://user-images.githubusercontent.com/60387522/175995430-76d525ab-de78-49f8-9a88-91befc86f0f6.png)

**Reply by Savage1969** [reply]
Good afternoon. I do have similar in a block. And that always worked just fine by itself of course:
`    "files.exclude": {
        "{**/*.rpa,**/*.rpyc}": true
    },`

But that excludes those extensions regardless. So, with that said what exactly does `"renpy.excludeRpycFilesFromWorkspace"` toggle?

Apologies for the confusion but I thought that the RenPy rpyc settings toggle either negated the "files.exclude" setting or it's use. So, I had experimented by commenting out that particular "files.exclude" block to test and obviously failed.

**Reply by duckdoom4** [reply]
Yeah, it's a very good point. I'm not sure why `"renpy.excludeRpycFilesFromWorkspace"` is a thing. It might be because it would otherwise not be possible to provide a config setting as part of the extension.

Anyway. Thanks to your question, I did figure out what caused the issue. Toggling `"renpy.excludeRpycFilesFromWorkspace"` does in fact, not work until you restart VSCode. I've pushed a modification that now updates the config based on the value set.

Edit: If it wasn't clear from the text above, the `"renpy.excludeRpycFilesFromWorkspace"` setting adds `"files.exclude": { "{**/*.rpa,**/*.rpyc}": true },` to the config

**Reply by duckdoom4** [reply]
Maybe @LuqueDaniel can answer that one?

**Reply by LuqueDaniel** [reply]
Yes, the purpose of `renpy.excludeRpycFilesFromWorkspace` was to add the config. Thanks @JelleInfinity for the fix.

**Reply by Savage1969** [reply]
Apologies, I never thought to restart VSC which would have been more helpful to isolate before my initial question. If I may suggest is to add the rpa extension as I did in my block:  `"files.exclude": { "{**/*.rpa,**/*.rpyc}": true }` since I doubt folks want those viewed if they don't want rpyc extensions viewed. Just my opinion though. Regards.

**Reply by duckdoom4** [reply]
No need to apologize, it was a bug and has been fixed now :)

I think there are some more files that should be hidden. Maybe the setting should be changed to 'excludeCompiledRenpyFilesFromWorkspace'.

As an example, along with `.rpa` there is `.rpymc`.

We then might also want to exclude the `cache` folder

**Reply by LuqueDaniel** [reply]
> Apologies, I never thought to restart VSC which would have been more helpful to isolate before my initial question. If I may suggest is to add the rpa extension as I did in my block: `"files.exclude": { "{**/*.rpa,**/*.rpyc}": true }` since I doubt folks want those viewed if they don't want rpyc extensions viewed. Just my opinion though. Regards.

Don't apologize. Thanks to your question we have been able to identify a bug.
